crates/store/README.md

```diff
@@ -1,163 +1,133 @@
 # Miden node store
 
-The **Store** maintains the state of the chain. It serves as the "source of truth" for the chain - i.e., if it is not in
-the store, the node does not consider it to be part of the chain.
-Incoming requests to the store are trusted because they are validated in the RPC component.
-
-**Store** is one of components of the [Miden node](..).
-
-## Architecture
-
-`TODO`
-
-## Usage
-
-### Installing the Store
-
-The Store can be installed and run as part of [Miden node](../README.md#installing-the-node).
-
-## API
-
-The **Store** serves connections using the [gRPC protocol](https://grpc.io) on a port, set in the previously mentioned configuration file. 
-Here is a brief description of supported methods.
+Contains the code defining the [Miden node's store component](/README.md#architecture). This component stores the
+network's state and acts as the networks source of truth. It serves a [gRPC](https://grpc.io) API which allow the other
+node components to interact with the store. This API is **internal** only and is considered trusted i.e. the node
+operator must take care that the store's API endpoint is **only** exposed to the other node components.
+
+For more information on the installation and operation of this component, please see the [node's readme](/README.md).
+
+## API overview
+
+The full gRPC API can be found [here](../../proto/store.proto).
+
+<!--toc:start-->
+- [ApplyBlock](#applyblock)
+- [CheckNullifiers](#checknullifiers)
+- [CheckNullifiersByPrefix](#checknullifiersbyprefix)
+- [GetAccountDetails](#getaccountdetails)
+- [GetAccountProofs](#getaccountproofs)
+- [GetAccountStateDelta](#getaccountstatedelta)
+- [GetBlockByNumber](#getblockbynumber)
+- [GetBlockHeaderByNumber](#getblockheaderbynumber)
+- [GetBlockInputs](#getblockinputs)
+- [GetNoteAuthenticationInfo](#getnoteauthenticationinfo)
+- [GetNotesById](#getnotesbyid)
+- [GetTransactionInputs](#gettransactioninputs)
+- [SyncNotes](#syncnotes)
+- [SyncState](#syncstate)
+<!--toc:end-->
+
+---
 
 ### ApplyBlock
 
 Applies changes of a new block to the DB and in-memory data structures.
```

Comment:
I would also maybe mention that this also saves the raw block data as a flat file.